File: src/Java/myservicebus-http/src/main/java/com/myservicebus/http/BusRegistrationContext.java
package com.myservicebus.http;

import com.myservicebus.di.ServiceProvider;

public class BusRegistrationContext {
    private final ServiceProvider serviceProvider;

    public BusRegistrationContext(ServiceProvider serviceProvider) {
        this.serviceProvider = serviceProvider;
    }

    public ServiceProvider getServiceProvider() {
        return serviceProvider;
    }
}

File: src/MyServiceBus/BusRegistrationContext.cs
namespace MyServiceBus;

public sealed class BusRegistrationContext : IBusRegistrationContext
{
    public BusRegistrationContext(IServiceProvider serviceProvider)
    {
        ServiceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
    }

    public IServiceProvider ServiceProvider { get; }
}
